authors/apps/authentication/tests/test_jwt_authentication.py

```diff
@@ -0,0 +1,80 @@
+from rest_framework import status
+
+class TestTokenAuthentication(BaseTest):
+    """
+    This class handles the testing of JWTAuthentication
+    """
+
+    def test_get_jwt_after_login(self):
+        self.signup_user('issa', 'issamwangi@gmail.com', 'Maina9176')
+        response = self.login_user('issamwangi@gmail.com', 'Maina9176')
+        self.assertIn('token', str(response.data))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_get_jwt_after_signup(self):
+        response = self.signup_user(
+            'issa', 'issamwangi@gmail.com', 'Maina9176')
+        self.assertIn('token', str(response.data))
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_invalid_authorization_header(self):
+        token = self.generate_jwt_token('issamwangi@gmail.com', 'issa')
+        self.client.credentials(
+            HTTP_AUTHORIZATION=token)
+        response = self.client.get(self.get_url)
+        self.assertEqual(
+            response.data['detail'],
+            'The authorization header provided is invalid!')
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_invalid_header_prefix(self):
+        token = self.generate_jwt_token('issamwangi@gmail.com', 'issa')
+        self.client.credentials(
+            HTTP_AUTHORIZATION='Invalid ' + token)
+        response = self.client.get(self.get_url)
+        self.assertEqual(
+            response.data['detail'],
+            'Please use a Bearer token!')
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_invalid_token(self):
+        token = 'sxdcfvgbhjklrty567dfg'
+        self.client.credentials(
+            HTTP_AUTHORIZATION='Bearer ' + token)
+        response = self.client.get(self.get_url)
+        self.assertEqual(
+            response.data['detail'],
+            'Invalid Token. The token provided cannot be decoded!')
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_inactive_user(self):
+        token = self.generate_jwt_token('ian@gmail.com', 'ian')
+        self.client.credentials(
+            HTTP_AUTHORIZATION='Bearer ' + token)
+        response = self.client.get(self.get_url)
+        self.assertEqual(
+            response.data['detail'],
+            'Your account is not active!')
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_user_not_found_from_token(self):
+        token = self.generate_jwt_token('ianmwangi@gmail.com', 'ianmaina')
+        self.client.credentials(
+            HTTP_AUTHORIZATION='Bearer ' + token)
+        response = self.client.get(self.get_url)
+        self.assertEqual(
+            response.data['detail'],
+            'No user was found from the provided token!')
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_expired_jwt_token(self):
+        token = os.getenv('EXPIREDTOKEN')
```

Comment:
Hello @IssaIan ,
If the expired token is an environment variable then you should update it in .env-example so that we could use it when running tests locally.
However, since we will not need it for developing features, you could explicitly declare it in the tests as a constant in the base test instead of having it set as an environment variable.

Reply:
@Ogutu-Brian, I have given this a thought. However, given that I am testing an expired jwt token that was created using my own secret key which is different from yours, this test will always fail locally on your machine. Making this token a constant within the test file is also a challenge because the token is too long and supersedes the pep8 standards. I was left with the option of adding the token as an environment variable. I could add the token to the .env-example file where you could copy  it. You will also need to have my secret key too in order for the test to pass. @oluwagbenga-joloko you can advice further on this issue.

Reply:
I think you should create a new token with a past `exp` value. This would ensure the test would pass on the machines where the secret key value is stored. 
```suggestion
        token = token = jwt.encode(
            {
                'user_data': user_details,
                'exp': datetime.now() - timedelta(hours=24)
            }, settings.SECRET_KEY, algorithm='HS256'
        )
```

Reply:
Thanks @oluwagbenga-joloko. I have implemented this. The tests should now pass locally @Ogutu-Brian with the new implementation.